nsxt/resource_nsxt_lb_service.go

```diff
@@ -0,0 +1,207 @@
+/* Copyright © 2018 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	api "github.com/vmware/go-vmware-nsxt"
+	"github.com/vmware/go-vmware-nsxt/loadbalancer"
+	"log"
+	"net/http"
+)
+
+var lbServiceLogLevels = []string{"DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL", "ALERT", "EMERGENCY"}
+var lbServiceSizes = []string{"SMALL", "MEDIUM", "LARGE"}
+
+func resourceNsxtLbService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtLbServiceCreate,
+		Read:   resourceNsxtLbServiceRead,
+		Update: resourceNsxtLbServiceUpdate,
+		Delete: resourceNsxtLbServiceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"revision": getRevisionSchema(),
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "Description of this resource",
+				Optional:    true,
+			},
+			"display_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The display name of this resource. Defaults to ID if not set",
+				Optional:    true,
+				Computed:    true,
+			},
+			"tag": getTagsSchema(),
+			"enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Description: "Whether the load balancer service is enabled",
+				Optional:    true,
+				Default:     true,
+			},
+			"error_log_level": &schema.Schema{
+				Type:         schema.TypeString,
+				Description:  "Load balancer engine writes information about encountered issues of different severity levels to the error log. This setting is used to define the severity level of the error log",
+				Optional:     true,
+				Default:      "INFO",
+				ValidateFunc: validation.StringInSlice(lbServiceLogLevels, false),
+			},
+			"size": &schema.Schema{
+				Type:         schema.TypeString,
+				Description:  "Size of load balancer service",
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(lbServiceSizes, false),
```

Comment:
Is this really 'required'? no default size?

Reply:
I think it might be helpful to force the user to think about size. But I can change it to "SMALL" if you think it's better.

Reply:
I think it is better to have the same default as the UI/backend